src/utils/polygon.cpp

```diff
@@ -156,6 +156,92 @@ Polygons PolygonRef::offset(int distance, ClipperLib::JoinType joinType, double
     return ret;
 }
 
+Polygon Polygons::convexHull() const
+{
+    // Implements Andrew's monotone chain convex hull algorithm
+    // See https://en.wikibooks.org/wiki/Algorithm_Implementation/Geometry/Convex_hull/Monotone_chain
+
+    using IntPoint = ClipperLib::IntPoint;
+
+    size_t num_points = 0U;
+    for (const ClipperLib::Path &path : paths)
+    {
+        num_points += path.size();
+    }
+
+    std::vector<IntPoint> all_points;
+    all_points.reserve(num_points);
```

Comment:
In the comments of previous commits we've discussed that this should be `num_points + 1`

Consider the case
```
.   .
.   .
```

In an intermediate step it would than have one of the four points twice, resulting in 5 points, which is one more than there are in the original poly.